docs/concepts.md

```diff
@@ -0,0 +1,56 @@
+# yb Concepts
+
+## Package
+
+A _package_ is a collection of related files and targets. A package is a
+directory that contains a .yourbase.yml file directly inside it. Files can only
+belong to a single package. A package may contain subpackages in subdirectories,
+but files within those subdirectories belong to the subpackage, not the parent
+package.
+
+## Target
+
+A _target_ is a buildable thing &mdash; it could be a service, a library,
+a binary, etc. A target has four _phases_ (all optional, but at least one must
+be present):
+
+-  Build
+-  Test
+-  Execute
+-  Release
```

Comment:
I'm super on board with being opinionated about these four phases existing as first-class citizens, and not just shoving them all into the "build" cubby 👍

However with _each target_ having each of these four phases, it's unclear to me how we expect customers to draw the lines between them. For example, take a project which in today's world has these targets:
```
seeddb
test
integrationtest
release
```
Plus this "non-target":
```
exec
```

With each target gaining build/test/execute/release, how do we expect this project to split their targets up? The closest I can see is something like
```
seeddb
  build
api
  build
  test
  exec
  release
integration
  test
```

And this seems like it's actually more shoehorned than the current system, not less.

Should we consider build/test/exec/release to be _categories of targets_, such that typing `yb test` runs all targets categorized as tests, `yb exec` runs all targets (e.g. PostgreSQL, Redis, app, etc.) categorized as `exec`, etc.?

As a meta-comment, it would be a good practice to go through some of our customers' build configs and refactor them to how they would look in this new system to reveal any oddness. Happy to pair on some of this for the benefit of creative juices.